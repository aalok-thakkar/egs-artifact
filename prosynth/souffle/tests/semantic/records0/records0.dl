/*
 * Souffle - A Datalog Compiler
 * Copyright (c) 2019 The Souffle Developers. All Rights reserved
 * Licensed under the Universal Permissive License v 1.0 as shown at:
 * - https://opensource.org/licenses/UPL
 * - <souffle root>/licenses/SOUFFLE-UPL.txt
 */

/*
 * Test handling of a zero length record
 */ 
.type empty = [ ]

.decl A, B(x:empty)

A([]).
A([]).

.printsize A, B
