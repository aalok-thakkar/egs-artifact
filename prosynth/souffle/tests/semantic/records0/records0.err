Warning: No rules/facts defined for relation B in file records0.dl at line 14
.decl A, B(x:empty)
---------^----------
