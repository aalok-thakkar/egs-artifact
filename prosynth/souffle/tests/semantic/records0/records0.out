A	1
B	0
